### DISCUSSION TITLE
Linstor thin volumes after reverting or creating from snapshot became a full size volume

### DISCUSSION THREAD

**Discussion by SviridoffA** [apache/cloudstack#9950] [Q&A] [answered]
Hi @rp- , maybe you or someone from community can help me. I have found starnge behavor on my infrastracture. I'm using
Ubuntu 24.04.1
Cloudstack 4.19.1.3
Linstor 1.29.1
LVM-THIN
When I reverting volume to snapshot state or when I create new volume from snapshot it allways became a full size volume instead of thin. Provisioning not changing, it's stays thin. Can someone give me advice, what i'm doing wrong?

**Comment by SviridoffA** [comment]
Also noticed that root disks created from templates are also full sized. Only if we creating disk from scratch its real thin

**Comment by rp-** [comment, marked as answer]
Hi!
These issues are because there is not yet, a released version of the following commits(4.19 branch):
27f23f4f75a4e19e014dcf986c59f2a365d3d716
7fbd0a9fb4d0a2f5764b9f0236d546915b6fea92
a6cef7a78ddbb83cdf5e207e1c79f0c79512a2ec

4.20 will have all of them and also a 4.19.2, or build your own version of plugins kvm and Linstor

**Reply by SviridoffA** [reply]
Thank you so much for the information, it is very useful.
